### PR TITLE
must create log group before creating/updating a statemachine that logs

### PR DESCRIPTION
**Pull request by rgarcia**
## Link to JIRA:
https://clever.atlassian.net/browse/INFRANG-4473

## Overview:
Follow up to previous [PR](https://github.com/Clever/workflow-manager/pull/317) that failed to do this necessary step.

If you made any changes to swagger.yml:
- [ ] Update swagger.yml version
- [ ] Run "make generate"

## Testing

Will do some local testing of this

### CODE REVIEW

**Comment by rgarcia** [comment]
tested successfully with `WORKFLOWMANAGER_URL=http://localhost:8080 ark submit -e clever-dev sfncli-dummy-worker:master '{"foo":true,"sleep_duration":"1s"}' `

Log group created and have some logs!
https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logsV2:log-groups/log-group/$252Fsfn$252Fclever-dev--sfncli-dummy-worker-master--2--start/log-events/states$252Fclever-dev--sfncli-dummy-worker-master--2--start$252F2022-08-24-16$252F7a175237

File: executor/workflow_manager_sfn.go [hunk 1]
```diff
@@ -176,7 +196,34 @@ func loggingConfiguration(logGroupARN string) *sfn.LoggingConfiguration {
...
 
-func (wm *SFNWorkflowManager) updateLoggingConfiguration(ctx context.Context, stateMachineARN string, lc *sfn.LoggingConfiguration) error {
+func (wm *SFNWorkflowManager) createLogGroupsForLoggingConfiguration(ctx context.Context, tags map[string]string, lc *sfn.LoggingConfiguration) error {
+	// for now there's only one log group destination in our state machine logging configuration, but loop over them for completeness
+	for _, ld := range lc.Destinations {
+		parts := strings.Split(*ld.CloudWatchLogsLogGroup.LogGroupArn, ":")
+		if len(parts) != 8 {
+			return fmt.Errorf("unexpected log group ARN format: %s", *ld.CloudWatchLogsLogGroup.LogGroupArn)
+		}
+		fmt.Println("DEBUG", *ld.CloudWatchLogsLogGroup.LogGroupArn)
```
Comment (tnsardesai) [inline executor/workflow_manager_sfn.go:206]: remove debug print statement

File: executor/workflow_manager_sfn.go [hunk 2]
```diff
@@ -176,7 +196,34 @@ func loggingConfiguration(logGroupARN string) *sfn.LoggingConfiguration {
...
+		lgname := parts[6]
+		fmt.Println("DEBUG", lgname)
```
Comment (tnsardesai) [inline executor/workflow_manager_sfn.go:208]: same

**Review by tnsardesai — APPROVED** [review]
🚢

File: executor/workflow_manager_sfn.go (hunk 1 above)
Comment (rgarcia) [reply executor/workflow_manager_sfn.go:206]: derp

**Pull request merged**